editoast/src/views/path/path_item_cache.rs

```diff
@@ -42,31 +51,68 @@ impl PathItemCache {
             return Ok(PathItemCache::default());
         }
 
+        // Step 1: Retrieve operational points from database using the requested identifiers
         let (trigrams, ops_uic, ops_id) = collect_path_item_ids(path_items);
-        let uic_to_ops = retrieve_op_from_uic(conn, infra_id, &ops_uic).await?;
-        let trigram_to_ops = retrieve_op_from_trigrams(conn, infra_id, &trigrams).await?;
-        let ids_to_ops = retrieve_op_from_ids(conn, infra_id, &ops_id).await?;
+        let uic_results = retrieve_op_from_uic(conn, infra_id, &ops_uic).await?;
+        let trigram_results = retrieve_op_from_trigrams(conn, infra_id, &trigrams).await?;
+        let ids_results = retrieve_op_from_ids(conn, infra_id, &ops_id).await?;
+
+        // Step 2: Collect all unique OPs first, deduplicating by obj_id
+        let id_to_op: HashMap<String, OperationalPointModel> = ids_results
```

Comment:
```suggestion
        let ops: Vec<OperationalPointModel> = ids_results
```

suggestion: build the OPs vector directly here so you can derive all indexes from it, including `id_to_op`

This way in your loop you can just `for (index, op) in ops.iter().enumerate()`

Reply:
Done in [ae1017364](https://github.com/OpenRailAssociation/osrd/compare/ae1017364c7e150bfd3de861a4be285227d42395..9342d451d5db7a51f31219e7dde37ebcbbfd0639)